standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/dbinstall/rules/Derby.java

```diff
@@ -88,7 +88,7 @@ public String getInitialJdbcUrl(String hostAddress) {
   }
 
   public String getDb() {
-    return MetaStoreServerUtils.JUNIT_DATABASE_PREFIX;
+    return "${test.tmp.dir}/" + MetaStoreServerUtils.JUNIT_DATABASE_PREFIX;
```

Comment:
I don't think this is necessary. First it is not going to be replaced like that and second it is used in conjunction with an in-memory derby so no DB file is created in the file-system.

Reply:
https://db.apache.org/derby/docs/10.8/devguide/cdevdvlp17453.html

Reply:
We already use the prefix for the in memory datastore in multiple places:
https://github.com/apache/hive/blob/2524c2137bc2e70bd32e43a49a15fbfc0fa93159/standalone-metastore/metastore-server/pom.xml#L395

https://github.com/apache/hive/blob/2524c2137bc2e70bd32e43a49a15fbfc0fa93159/standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/MetaStoreFactoryForTests.java#L90-L92

https://github.com/apache/hive/blob/2524c2137bc2e70bd32e43a49a15fbfc0fa93159/standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/utils/TestMetaStoreServerUtils.java#L841-L842

I am not saying that this is good, I just state that we already use this several times already 😄.
Had to remove this change anyway, since the test failures for `TestMiniLlapLocalCliDriver`, `TestCliDriver`, ... are caused by this and the fix is not straightforward.

Reply:
I wasn't aware of those usages but for sure we can do much better than this. Since you reverted the change anyways we can discuss this separately.